### PR TITLE
Removed override variables for lms/cms

### PR DESCRIPTION
**Pull request by syedimranhassan**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

File: playbooks/roles/edxapp/tasks/service_variant_config.yml [hunk 1]
```diff
@@ -1,80 +1,80 @@
...
-  no_log: "{{ COMMON_CONFIG_NO_LOGGING }}"
-  with_items: "{{ service_variants_enabled }}"
-  when: '"lms" in service_variants_enabled and not EDXAPP_DECRYPT_CONFIG_ENABLED'
-  tags:
-    - install
-    - install:configuration
-    - install:app-configuration
-    - edxapp_cfg # Old deprecated tag, will remove when possible
-    - edxapp_cfg_yaml_only # Used to render the yaml without the json until we remove the json configs
+#- name: create application and auth config
```
Comment (coryleeio) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml]: You can remove this, no need to leave the commented code

File: playbooks/roles/edxapp/tasks/service_variant_config.yml [hunk 2]
```diff
@@ -1,80 +1,80 @@
...
-    cms_combined_config: '{{cms_env_config|combine(cms_auth_config)}}'
-  no_log: "{{ COMMON_CONFIG_NO_LOGGING }}"
-  when: '"cms" in service_variants_enabled and not EDXAPP_DECRYPT_CONFIG_ENABLED'
-  tags:
-    - install
-    - install:configuration
-    - install:app-configuration
-    - edxapp_cfg # Old deprecated tag, will remove when possible
-    - edxapp_cfg_yaml_only # Used to render the yaml without the json until we remove the json configs
+#- name: combined lms auth env for yml
```
Comment (coryleeio) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml]: go ahead and remove

File: playbooks/roles/edxapp/tasks/service_variant_config.yml [hunk 3]
```diff
@@ -1,80 +1,80 @@
...
-  no_log: "{{ COMMON_CONFIG_NO_LOGGING }}"
-  with_items: "{{ service_variants_enabled }}"
-  when: '"cms" in service_variants_enabled and not EDXAPP_DECRYPT_CONFIG_ENABLED'
-  tags:
-    - install
-    - install:configuration
-    - install:app-configuration
-    - edxapp_cfg # Old deprecated tag, will remove when possible
-    - edxapp_cfg_yaml_only # Used to render the yaml without the json until we remove the json configs
+#- name: combined cms auth env for yml
```
Comment (coryleeio) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml]: remove - we have this in git history,  no reason to keep it

File: playbooks/roles/edxapp/tasks/service_variant_config.yml [hunk 4]
```diff
@@ -211,19 +134,6 @@
     - install
     - install:configuration
 
-- name: create helper scripts for managing edxapp
```
Comment (coryleeio) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml]: I think we may need to leave these - I see references to them in the documentation and the presupervisor checks

**Review by coryleeio — APPROVED** [review]
I'm thumbing so you can keep moving forward as you have the right idea - I think you may need to review my last comment and undo the deletion of the migrate scripts though

File: playbooks/roles/edxapp/tasks/service_variant_config.yml (hunk 4 above)
Comment (syedimranhassan) [reply playbooks/roles/edxapp/tasks/service_variant_config.yml]: After adding this task test are failing with these errors
`"AnsibleUndefinedVariable: 'lms_auth_config' is undefined"`
`"AnsibleUndefinedVariable: 'cms_auth_config' is undefined"`
We had removed these variables in this PR.

**Review by coryleeio — APPROVED** [review]

**Review by coryleeio — CHANGES_REQUESTED** [review]
This looks good but we can't actually merge it until we get devstack off of the rendered ansible configuration.

I have created https://openedx.atlassian.net/browse/OPS-4397 to represent this.

**Review by coryleeio — APPROVED** [review]

**Comment by pomegranited** [comment]
Hi @syedimranhassan CC @coryleeio This PR removes the default values for [`EDXAPP_MEMCACHE`](https://github.com/edx/configuration/commit/9920561bffe1f3a3922caeedfe6e6d6ba0017484#diff-9bffb77ca37dcc0aead699b822b29095L144) and [`EDXAPP_ELASTIC_SEARCH_CONFIG`](https://github.com/edx/configuration/commit/9920561bffe1f3a3922caeedfe6e6d6ba0017484#diff-9bffb77ca37dcc0aead699b822b29095L122-L125) and so breaks the playbooks that assume they are defined: [edx-stateless.yml](https://github.com/edx/configuration/blob/cf4d221d384ed396a3008c31487f385544ef08e7/playbooks/edx-stateless.yml) and [openedx_native.yml](https://github.com/edx/configuration/blob/70217a3ded0f623b6ad98218cde3ce46e956c8cc/playbooks/openedx_native.yml).

Could these playbooks please be fixed?

**Comment by syedimranhassan** [comment]
Hi @pomegranited We are working on this fix. The PR is in progress. Hopefully, we will merge it by today.

**Comment by pomegranited** [comment]
Awesome, thank you for getting on this so quickly, @syedimranhassan !

**Comment by bradenmacdonald** [comment]
@syedimranhassan CC @coryleeio Just want to make sure you've seen my comment at https://github.com/edx/configuration/pull/5414#issuecomment-588398918 - we have some concerns about this approach, and I'm hoping you might have a solution in mind that I haven't thought of yet.

**Pull request merged**